src/runtime/security/security.thrift

```diff
@@ -0,0 +1,46 @@
+include "../../dsn.thrift"
+
+namespace cpp dsn.security
+
+// negotiation process:
+//
+//                       client                              server
+//                          | ---   SASL_LIST_MECHANISMS    --> |
+//                          | <-- SASL_LIST_MECHANISMS_RESP --- |
+//                          | --    SASL_SELECT_MECHANISMS  --> |
+//                          | <-- SASL_SELECT_MECHANISMS_OK --- |
+//                          |                                   |
+//                          | ---      SASL_INITIATE        --> |
+//                          |                                   |
+//                          | <--      SASL_CHALLENGE       --- |
+//                          | ---  SASL_CHALLENGE_RESPONSE  --> |
+//                          |                                   |
+//                          |              .....                |
+//                          |                                   |
+//                          | <--      SASL_CHALLENGE       --- |
+//                          | ---  SASL_CHALLENGE_RESPONSE  --> |
+//                          |                                   | (authentication will succeed
+//                          |                                   |  if all chanllenges passed)
+//                          | <--        SASL_SUCC          --- |
+// (client won't response   |                                   |
+// if servers says ok)      |                                   |
+//                          | ---        RPC_CALL          ---> |
+//                          | <--        RPC_RESP          ---- |
+
+enum negotiation_status {
+    INVALID
+    SASL_LIST_MECHANISMS
+    SASL_LIST_MECHANISMS_RESP
+    SASL_SELECT_MECHANISMS
+    SASL_SELECT_MECHANISMS_OK
```

Comment:
What is the usage of four `LIST_MECHANISMS` status?

Reply:
In this status, client reuqest server to list the mechanism it supports, and select one the client support also to use